sdk/src/test/java/com/google/cloud/dataflow/sdk/transforms/DoFnTest.java

```diff
@@ -188,4 +193,30 @@ public void processElement(DoFn<String, String>.ProcessContext c) throws Excepti
 
     return pipeline;
   }
+  
+  @Test
+  public void testPopulateDisplayDataDefaultBehavior() {
+    DoFn<String, String> usesDefault =
+        new DoFn<String, String>() {
+          @Override
+          public void processElement(DoFn<String, String>.ProcessContext c) throws Exception {}
+        };
+
+    DisplayData data = collectDisplayData(usesDefault);
+    assertThat(data.items(), empty());
+  }
+
+  private DisplayData collectDisplayData(final DoFn<String, String> fn) {
```

Comment:
Yeah, it's definitely a lot of boilerplate as-is. I wanted to prevent people from misusing the API by limiting the accepted parameter: DisplayData is registered on a __PTransform__, and that transform may include it's on components.

I'm thinking we could have a `DisplayDataTester`, which has some useful wrapper APIs for testing.